File: 8_kyu/Get_Planet_Name_By_ID.go
package kata

var PlanetNames = [...]string{
  "Mercury", "Venus", "Earth",
  "Mars", "Jupiter", "Saturn",
  "Uranus", "Neptune", "Pluto" }

func GetPlanetName(id int) string {
    if id < 1 || id > len(PlanetNames) {
        id = len(PlanetNames) - 1
    } else {
        id = id - 1
    }
    return PlanetNames[id];
}

File: 8_kyu/Get_Planet_Name_By_ID.rs
fn get_planet_name(id: u32) -> String {
    match id {
        1 => "Mercury",
        2 => "Venus",
        3 => "Earth",
        4 => "Mars",
        5 => "Jupiter",
        6 => "Saturn",
        7 => "Uranus",
        8 => "Neptune",
        _ => unreachable!(),
    }
    .to_string()
}
